packages/aws-appsync/src/client.js

```diff
@@ -28,51 +84,47 @@ class AWSAppSyncClient extends ApolloClient {
 
     /**
      *
-     * @param {string} url
+     * @param {object} appSyncOptions
      * @param {ApolloClientOptions<InMemoryCache>} options
      */
-    constructor({ url, region, auth, conflictResolver, complexObjectsCredentials, disableOffline = false }, options) {
-        if (!url || !region || !auth) {
+    constructor({
+        url,
+        region,
+        auth,
+        conflictResolver,
+        complexObjectsCredentials,
+        cacheOptions,
+        disableOffline = false
+    } = {}, options = {}) {
+        const { cache: customCache, link: customLink } = options;
+
+        if (!customLink && (!url || !region || !auth)) {
             throw new Error(
                 'In order to initialize AWSAppSyncClient, you must specify url, region and auth properties on the config object.'
             );
         }
 
-        let res;
-        this.hydratedPromise = new Promise((resolve, reject) => {
-            res = resolve;
-        });
+        let resolveClient;
+
+        const store = disableOffline ? null : createStore(() => this, () => resolveClient(this), conflictResolver);
```

Comment:
I would like to talk tomorrow the use of ```resolveClient```

Reply:
Sure!

Reply:
@manueliglesias I don't have reproducible steps yet, but I can sometimes trigger an error in my app around this new area of the appsync sdk code:
TypeError: resolveClient is not a function. (In \'resolveClient(_this)\', \'resolveClient\' is undefined)")
It seems to be thrown when I invoke a lambda outside the context of appsync, while the app is starting up. Sorry I don't have more actionable information yet.

Reply:
When disableOffline=false, I applied this workaround to prevent the "resolveClient is not a function" error: https://github.com/apollographql/apollo-link/issues/493#issuecomment-372307620